### PR TITLE
Promote google_dataproc_metastore_service and related fields in google_dataproc_cluster to GA

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes https://github.com/hashicorp/terraform-provider-google/issues/9938

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
metastore: promote `google_dataproc_metastore_service` to GA
```

```release-note:enhancement
metastore: add `databaseType`, `releaseChannel`, and `hiveMetastoreConfig.endpointProtocol` arguments
```

### CODE REVIEW

File: mmv1/products/metastore/api.yaml [hunk 1]
```diff
@@ -169,6 +170,16 @@ objects:
         description: |
          Configuration information specific to running Hive metastore software as the metastore service.
         properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'endpointProtocol'
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply mmv1/products/metastore/api.yaml:174]: sure, will do

File: mmv1/products/metastore/terraform.yaml [hunk 2]
```diff
@@ -34,13 +33,18 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
       - !ruby/object:Provider::Terraform::Examples
         name: "dataproc_metastore_service_cmek_example"
-        min_version: beta
         skip_test: true
         primary_resource_id: "default"
         vars:
           metastore_service_name: "example-service"
           key_name: "example-key"
           keyring_name: "example-keyring"
+      - !ruby/object:Provider::Terraform::Examples
```
Comment (megan07) [inline mmv1/products/metastore/terraform.yaml:42]: I believe we'll want this test to be `min_version: beta` correct?

**Comment by DrFaust92** [comment]
found some issues with tests, checking

**Comment by DrFaust92** [comment]
should be passing now @megan07

**Review by megan07 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**